src/dir_entry.rs

```diff
@@ -31,6 +40,40 @@ impl<'a> Debug for DirEntryName<'a> {
     }
 }
 
+impl<'a> TryFrom<&'a [u8]> for DirEntryName<'a> {
+    type Error = DirEntryNameError;
```

Comment:
I'm not clear what purpose these type aliases are serving here, nor why they need to be duplicated in all the impl blocks.

Reply:
`type X = Y` as a top-level item is a type alias, but has a different meaning in a trait impl: https://doc.rust-lang.org/rust-by-example/generics/assoc_items/types.html.

For the `TryFrom` trait, `Error` is an arbitrary type that the implementor must provide.